packages/go/cypher/backend/pgsql/format.go

```diff
@@ -240,207 +240,175 @@ func (s *Emitter) writeWhere(writer io.Writer, whereClause *model.Where) error {
-		if nodePattern, isNodePattern := patternElement.AsNodePattern(); isNodePattern {
-			if idx == 0 {
-				if _, err := io.WriteString(writer, pgDriverModel.NodeTable); err != nil {
-					return nil
-				}
-
-				if _, err := io.WriteString(writer, " as "); err != nil {
-					return nil
-				}
-
-				if err := s.WriteExpression(writer, nodePattern.Binding); err != nil {
-					return nil
-				}
-			} else {
-				previousRelationshipPattern, _ := patternElements[idx-1].AsRelationshipPattern()
-
-				if _, err := WriteStrings(writer, " join ", pgDriverModel.NodeTable, " "); err != nil {
-					return err
-				}
-
-				if err := s.WriteExpression(writer, nodePattern.Binding); err != nil {
-					return err
-				}
-
-				if _, err := WriteStrings(writer, " on "); err != nil {
-					return err
-				}
-
-				switch previousRelationshipPattern.Direction {
-				case graph.DirectionOutbound:
-					if err := s.WriteExpression(writer, nodePattern.Binding); err != nil {
-						return err
-					}
-
-					if _, err := WriteStrings(writer, ".id = "); err != nil {
-						return err
-					}
-
-					if err := s.WriteExpression(writer, previousRelationshipPattern.Binding); err != nil {
-						return err
-					}
-
-					if _, err := WriteStrings(writer, ".end_id"); err != nil {
-						return err
-					}
-
-				case graph.DirectionInbound:
-					if err := s.WriteExpression(writer, nodePattern.Binding); err != nil {
-						return err
-					}
-
-					if _, err := WriteStrings(writer, ".id = "); err != nil {
-						return err
-					}
-
-					if err := s.WriteExpression(writer, previousRelationshipPattern.Binding); err != nil {
-						return err
-					}
-
-					if _, err := WriteStrings(writer, ".start_id"); err != nil {
-						return err
-					}
+const traversalHeader = `with recursive `
+const traversalTablePrefix = `pathspace_`
+const traversalTableDef = `(next_node_id, depth, is_cycle, path) as (`
+
+func expressionAsType[T any](expr model.Expression) (T, error) {
+	if typed, isType := expr.(T); !isType {
+		var empty T
+		return empty, fmt.Errorf("expected type %T but received %T", empty, expr)
+	} else {
+		return typed, nil
+	}
+}
 
-				default:
-					if err := s.WriteExpression(writer, nodePattern.Binding); err != nil {
-						return err
-					}
+func formatVariableExpression(expr model.Expression, defaultBinding string) (string, error) {
+	if expr == nil {
+		return defaultBinding, nil
+	}
 
-					if _, err := WriteStrings(writer, ".id = "); err != nil {
-						return err
-					}
+	if variable, err := expressionAsType[*pgModel.AnnotatedVariable](expr); err != nil {
+		return "", err
+	} else {
+		return variable.Symbol, nil
+	}
+}
 
-					if err := s.WriteExpression(writer, previousRelationshipPattern.Binding); err != nil {
-						return err
-					}
+const (
+	defaultNodeBinding = "n"
+	defaultEdgeBinding = "r"
+)
 
-					if _, err := WriteStrings(writer, ".start_id or "); err != nil {
-						return err
-					}
+// ()-[]
+// ()<-[]
```

Comment:
Without much of an explanation these comments just look out of place. Could you elaborate on how this relates to the method below or remove it?

Reply:
These are just old cruft and part of code paths that will eventually be deprecated so expect removal in a future changeset